Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
+
+    Check.typeOf = {};
+    Check.numeric = {};
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return defaultValue(name, 'a value') + ' is required but is undefined.';
+    }
+    function getFailedTypeErrorMessage(actual, expected) {
+        return 'Expected ' + expected + ', got ' + actual;
+    }
+
+    /**
+     * @param {} test The value that is to be checked
+     * @param {string} name The name of the variable being tested
+     * @throws {DeveloperError}
+     */
+    Check.defined = function (test, name) {
+        if (!defined(test)) {
+            throw new DeveloperError(errors.defined(name));
+        }
+    };
+
+    /**
+     * @param {} test The value to test
+     * @param {number} min The minimum allowed value
+     * @param {number} max The maximum allowed value
+     * @throws {DeveloperError}
+     */
+    Check.numeric.withinRange = function (test, min, max) {
+        Check.typeOf.number(test);
+        Check.typeOf.number(max);
+        Check.typeOf.number(min);
+        if (min > max) {
```

Comment:
Yeah, it's probably better to leave for when it comes up in practice, it doesn't seem Cesium has a lot of checks of this kind anyway.